File: Database/Assembly_SRPG_JP/RarityParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.RarityParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

namespace SRPG
{
  public class RarityParam
  {
    public static readonly int MAX_RARITY = 6;
    public StatusParam GrowStatus = new StatusParam();
    public OInt UnitLvCap;
    public OInt UnitJobLvCap;
    public OInt UnitAwakeLvCap;
    public OInt UnitUnlockPieceNum;
    public OInt UnitChangePieceNum;
    public OInt UnitSelectChangePieceNum;
    public OInt UnitRarityUpCost;
    public OInt PieceToPoint;
    public string DropSE;
    public RarityEquipEnhanceParam EquipEnhanceParam;
    public OInt ArtifactLvCap;
    public OInt ArtifactCostRate;
    public OInt ArtifactCreatePieceNum;
    public OInt ArtifactGouseiPieceNum;
    public OInt ArtifactChangePieceNum;
    public OInt ArtifactCreateCost;
    public OInt ArtifactRarityUpCost;
    public OInt ArtifactChangeCost;
    public OInt ConceptCardLvCap;
    public OInt ConceptCardAwakeCountMax;

    public bool Deserialize(JSON_RarityParam json)
    {
      if (json == null)
        return false;
      this.UnitLvCap = (OInt) json.unitcap;
      this.UnitJobLvCap = (OInt) json.jobcap;
      this.UnitAwakeLvCap = (OInt) json.awakecap;
      this.UnitUnlockPieceNum = (OInt) json.piece;
      this.UnitChangePieceNum = (OInt) json.ch_piece;
      this.UnitSelectChangePieceNum = (OInt) json.ch_piece_select;
      this.UnitRarityUpCost = (OInt) json.rareup_cost;
      this.PieceToPoint = (OInt) json.gain_pp;
      if (this.EquipEnhanceParam == null)
        this.EquipEnhanceParam = new RarityEquipEnhanceParam();
      int length = json.eq_enhcap + 1;
      this.EquipEnhanceParam.rankcap = (OInt) length;
      this.EquipEnhanceParam.cost_scale = (OInt) json.eq_costscale;
      this.EquipEnhanceParam.ranks = (RarityEquipEnhanceParam.RankParam[]) null;
      if (length > 0)
      {
        if (json.eq_points == null || json.eq_num1 == null || (json.eq_num2 == null || json.eq_num3 == null))
          return false;
        this.EquipEnhanceParam.ranks = new RarityEquipEnhanceParam.RankParam[length];
        for (int index = 0; index < length; ++index)
        {
          this.EquipEnhanceParam.ranks[index] = new RarityEquipEnhanceParam.RankParam();
          this.EquipEnhanceParam.ranks[index].need_point = (OInt) json.eq_points[index];
        }
        string[] strArray = new string[3]
        {
          json.eq_item1,
          json.eq_item2,
          json.eq_item3
        };
        int[][] numArray = new int[3][]
        {
          json.eq_num1,
          json.eq_num2,
          json.eq_num3
        };
        for (int index1 = 0; index1 < strArray.Length; ++index1)
        {
          for (int index2 = 0; index2 < length; ++index2)
          {
            this.EquipEnhanceParam.ranks[index2].return_item[index1] = new ReturnItem();
            this.EquipEnhanceParam.ranks[index2].return_item[index1].iname = strArray[index1];
            this.EquipEnhanceParam.ranks[index2].return_item[index1].num = (OInt) numArray[index1][index2];
          }
        }
      }
      this.ArtifactLvCap = (OInt) json.af_lvcap;
      this.ArtifactCostRate = (OInt) json.af_upcost;
      this.ArtifactCreatePieceNum = (OInt) json.af_unlock;
      this.ArtifactGouseiPieceNum = (OInt) json.af_gousei;
      this.ArtifactChangePieceNum = (OInt) json.af_change;
      this.ArtifactCreateCost = (OInt) json.af_unlock_cost;
      this.ArtifactRarityUpCost = (OInt) json.af_gousei_cost;
      this.ArtifactChangeCost = (OInt) json.af_change_cost;
      this.GrowStatus.hp = (OInt) json.hp;
      this.GrowStatus.mp = (OShort) json.mp;
      this.GrowStatus.atk = (OShort) json.atk;
      this.GrowStatus.def = (OShort) json.def;
      this.GrowStatus.mag = (OShort) json.mag;
      this.GrowStatus.mnd = (OShort) json.mnd;
      this.GrowStatus.dex = (OShort) json.dex;
      this.GrowStatus.spd = (OShort) json.spd;
      this.GrowStatus.cri = (OShort) json.cri;
      this.GrowStatus.luk = (OShort) json.luk;
      this.DropSE = json.drop;
      this.ConceptCardLvCap = (OInt) json.card_lvcap;
      this.ConceptCardAwakeCountMax = (OInt) json.card_awake_count;
      return true;
    }
  }
}

File: Database/genConvert/RarityParam.py
def RarityParam(json):
    this={}#RarityParamjson)
    #if(json==null)
    #returnfalse
    if 'unitcap' in json:
        this['UnitLvCap'] = json['unitcap']
    if 'jobcap' in json:
        this['UnitJobLvCap'] = json['jobcap']
    if 'awakecap' in json:
        this['UnitAwakeLvCap'] = json['awakecap']
    if 'piece' in json:
        this['UnitUnlockPieceNum'] = json['piece']
    if 'ch_piece' in json:
        this['UnitChangePieceNum'] = json['ch_piece']
    if 'ch_piece_select' in json:
        this['UnitSelectChangePieceNum'] = json['ch_piece_select']
    if 'rareup_cost' in json:
        this['UnitRarityUpCost'] = json['rareup_cost']
    if 'gain_pp' in json:
        this['PieceToPoint'] = json['gain_pp']
    #if(this.EquipEnhanceParam==null)
    #this.EquipEnhanceParam=newRarityEquipEnhanceParam()
    #intlength=json.eq_enhcap+1
    #this.EquipEnhanceParam.rankcap=(OInt)length
    this['']
    this['EquipEnhanceParam']
    if 'eq_costscale' in json:
        this['EquipEnhanceParam']['cost_scale'] = json['eq_costscale']
    #this.EquipEnhanceParam.ranks=(RarityEquipEnhanceParam.RankParam)null
    #if(length>0)
        #if(json.eq_points==null||json.eq_num1==null||(json.eq_num2==null||json.eq_num3==null))
        #returnfalse
        #this.EquipEnhanceParam.ranks=newRarityEquipEnhanceParam.RankParam[length]
        #for(intindex=0index<length++index)
            #this.EquipEnhanceParam.ranks=newRarityEquipEnhanceParam.RankParam()
            this['EquipEnhanceParam']
            this['EquipEnhanceParam']['ranks']
            if 'eq_points' in json:
                this['EquipEnhanceParam']['ranks']['need_point'] = json['eq_points']
        #stringstrArray=newstring[3]
            #json.eq_item1,
            #json.eq_item2,
            #json.eq_item3
            #}
            #intnumArray=newint[3]
                #json.eq_num1,
                #json.eq_num2,
                #json.eq_num3
                #}
                #for(intindex1=0index1<strArray.Length++index1)
                    #for(intindex2=0index2<length++index2)
                        #this.EquipEnhanceParam.ranks[index2].return_item[index1]=newReturnItem()
                        #this.EquipEnhanceParam.ranks[index2].return_item[index1].iname=strArray[index1]
                        #this.EquipEnhanceParam.ranks[index2].return_item[index1].num=(OInt)numArray[index1][index2]
            if 'af_lvcap' in json:
                this['ArtifactLvCap'] = json['af_lvcap']
            if 'af_upcost' in json:
                this['ArtifactCostRate'] = json['af_upcost']
            if 'af_unlock' in json:
                this['ArtifactCreatePieceNum'] = json['af_unlock']
            if 'af_gousei' in json:
                this['ArtifactGouseiPieceNum'] = json['af_gousei']
            if 'af_change' in json:
                this['ArtifactChangePieceNum'] = json['af_change']
            if 'af_unlock_cost' in json:
                this['ArtifactCreateCost'] = json['af_unlock_cost']
            if 'af_gousei_cost' in json:
                this['ArtifactRarityUpCost'] = json['af_gousei_cost']
            if 'af_change_cost' in json:
                this['ArtifactChangeCost'] = json['af_change_cost']
            this['']
            this['GrowStatus']
            if 'hp' in json:
                this['GrowStatus']['hp'] = json['hp']
            this['GrowStatus']
            if 'mp' in json:
                this['GrowStatus']['mp'] = json['mp']
            this['GrowStatus']
            if 'atk' in json:
                this['GrowStatus']['atk'] = json['atk']
            this['GrowStatus']
            if 'def' in json:
                this['GrowStatus']['def'] = json['def']
            this['GrowStatus']
            if 'mag' in json:
                this['GrowStatus']['mag'] = json['mag']
            this['GrowStatus']
            if 'mnd' in json:
                this['GrowStatus']['mnd'] = json['mnd']
            this['GrowStatus']
            if 'dex' in json:
                this['GrowStatus']['dex'] = json['dex']
            this['GrowStatus']
            if 'spd' in json:
                this['GrowStatus']['spd'] = json['spd']
            this['GrowStatus']
            if 'cri' in json:
                this['GrowStatus']['cri'] = json['cri']
            this['GrowStatus']
            if 'luk' in json:
                this['GrowStatus']['luk'] = json['luk']
            if 'drop' in json:
                this['DropSE'] = json['drop']
            if 'card_lvcap' in json:
                this['ConceptCardLvCap'] = json['card_lvcap']
            if 'card_awake_count' in json:
                this['ConceptCardAwakeCountMax'] = json['card_awake_count']
            #returntrue
return this
